llvm/lib/Transforms/Instrumentation/MemProfUse.cpp

```diff
@@ -798,13 +798,15 @@ bool MemProfUsePass::annotateGlobalVariables(
     return false;
 
   if (!DataAccessProf) {
+    M.addModuleFlag(Module::Warning, "EnableDataAccessProf", 0U);
```

Comment:
What if the function doesn't have any variables that get data access profiles? I assume we have an non-missing but empty DataAccessProfiles section so we wouldn't come in here?

Reply:
Thanks for the explanation, this makes sense.

> I wonder if it's clearer to combine 1) and 2) so the module flag is always present as long as this pass runs.

I don't think you want to do that, because a module without any globals would get a flag value of 0, and per Module::Warning semantics, if it is combined with another module via an LTO link the first value would win. I think you want this to emit the same value, or none, for all modules using the same profile.